File: gitdot-api/src/endpoint/review/get_reviews.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct GetReviews;

impl Endpoint for GetReviews {
    const PATH: &'static str = "/repository/{owner}/{repo}/reviews";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetReviewsRequest;
    type Response = GetReviewsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetReviewsRequest;

pub type GetReviewsResponse = Vec<ReviewResource>;

File: gitdot-web/packages/api/src/endpoint/review/get-reviews.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetReviewsRequest = z.object({});
export type GetReviewsRequest = z.infer<typeof GetReviewsRequest>;

export const GetReviewsResponse = z.array(ReviewResource);
export type GetReviewsResponse = z.infer<typeof GetReviewsResponse>;

export const GetReviews = {
  path: "/repository/{owner}/{repo}/reviews",
  method: "GET",
  request: GetReviewsRequest,
  response: GetReviewsResponse,
} as const satisfies Endpoint;
export type GetReviews = typeof GetReviews;
